build.gradle

```diff
@@ -20,9 +20,11 @@ ext {
   supportAnnotations = 'com.android.support:support-annotations:23.0.0'
   supportTestRunner = 'com.android.support.test:runner:0.3'
 
-  // Square dependencies.
-  dagger = 'com.squareup.dagger:dagger:1.2.2'
-  daggerCompiler = 'com.squareup.dagger:dagger-compiler:1.2.2'
+  // Dagger dependencies.
+  dagger = 'com.google.dagger:dagger:2.2'
+  daggerCompiler = 'com.google.dagger:dagger-compiler:2.2'
+  javaxAnnotations = 'javax.annotation:jsr250-api:1.0'
```

Comment:
This shouldn't be needed. Dagger won't include `@Generated` when it detects the type is absent from the classpath.